src/Microsoft.OData.Core/ODataMetadataOutputContext.cs

```diff
@@ -127,8 +179,18 @@ protected override void Dispose(bool disposing)
             {
                 if (this.xmlWriter != null)
                 {
-                    // XmlWriter.Flush will call the underlying Stream.Flush.
-                    this.xmlWriter.Flush();
+                    // In the async case the underlying stream is the async buffered stream, so we have to flush that explicitly.
+                    if (this.asynchronousOutputStream != null)
+                    {
+                        this.xmlWriter.FlushAsync();
+                        this.asynchronousOutputStream.FlushSync();

```

Comment:
>FlushSync [](start = 54, length = 9)

Why are we calling FlushSync, rather than FlushAsync, on the asynchronous stream?